config/initializers/application_record_serializing.rb

```diff
@@ -0,0 +1,17 @@
+ar_serializing = Module.new do
+  def as_json(arg = {})
+    previous = super
+    return previous unless arg.key?(:prefixes)
+    /\/v(?<major>\d+)[x\.]?(?<minor>\d+)?\// =~ arg[:prefixes].first
+    version = [major, minor].compact.join(".")
+    schema  = Api::Docs[version].definitions[self.class.name]
+    attrs   = previous.slice(*schema["properties"].keys)
+    schema["properties"].keys.each do |name|
+      attrs[name] = attrs[name].iso8601 if attrs[name].kind_of?(Time)
+      attrs[name] = attrs[name].to_s if name.ends_with?("_id") || name == "id"
+    end
+    attrs
+  end
+end
+
+ApplicationRecord.prepend(ar_serializing)
```

Comment:
Why not just include this in ApplicationRecord directly?

Reply:
ApplicationRecord should live with the models in the core repo and this is very API specific.  I think that this will need to be updated to handle serializing differently for multiple versions (another thing that is not the concern of the models), so I'd rather keep it here.